pravega/pravega
# Issue 279: Segment Store Metrics

**Change log description**
Added or updated metrics for a lot of SegmentStore operations on the write path:
- BookKeeper
    - BK Write Latency & Bytes
    - BookKeeperLog Latency (including queue time)
    - BookKeeperLog queue size and fill rate
- SegmentStore
    - OperationLog size
    - OperationProcessor metrics: various latencies, counts and other metrics useful for debugging
    - Cache: total size and generation spread.

**Purpose of the change**
Fixes #279.

**What the code does**
Reports metrics.

**How to verify it**
Enable metrics via CSV or Console and run a SelfTest using the SegmentStore; verify these metrics output data.
